python/pyspark/ml/common.py

```diff
@@ -111,13 +114,16 @@ def _java2py(sc, r, encoding="bytes"):
     return r
 
 
-def callJavaFunc(sc, func, *args):
+def callJavaFunc(sc: pyspark.context.SparkContext, func, *args):
     """Call Java Function"""
-    args = [_py2java(sc, a) for a in args]
-    return _java2py(sc, func(*args))
+    java_args = [_py2java(sc, a) for a in args]
+    return _java2py(sc, func(*java_args))
+
+
+C = TypeVar("C", bound=type)
```

Comment:
I'm not really sure that this is appropriate, since `C` now becomes part of the public interface for common.py:

```python
>>> from pyspark.ml.common import C
>>> C
~C
```

@zero323?

Reply:
Type looks reasonable, but `_typing.pyi` might be a better place for it.